src/NzbDrone.Core/Download/Clients/NzbVortex/NzbVortex.cs

```diff
@@ -0,0 +1,263 @@
+using NzbDrone.Core.Parser.Model;
+using NzbDrone.Core.Validation;
+using NzbDrone.Core.RemotePathMappings;
+
+namespace NzbDrone.Core.Download.Clients.NzbVortex
+{
+    public class NzbVortex : UsenetClientBase<NzbVortexSettings>
+    {
+        private readonly INzbVortexProxy _proxy;
+
+        public NzbVortex(INzbVortexProxy proxy,
+                       IHttpClient httpClient,
+                       IConfigService configService,
+                       IDiskProvider diskProvider,
+                       IRemotePathMappingService remotePathMappingService,
+                       Logger logger)
+            : base(httpClient, configService, diskProvider, remotePathMappingService, logger)
+        {
+            _proxy = proxy;
+        }
+
+        protected override string AddFromNzbFile(RemoteEpisode remoteEpisode, string filename, byte[] fileContent)
+        {
+            var priority = remoteEpisode.IsRecentEpisode() ? Settings.RecentTvPriority : Settings.OlderTvPriority;
+
+            var response = _proxy.DownloadNzb(fileContent, filename, priority, Settings);
+
+            if (response == null)
+            {
+                throw new DownloadClientException("Failed to add nzb {0}", filename);
+            }
+
+            return response;
+        }
+
+        public override string Name
+        {
+            get
+            {
+                return "NZBVortex";
+            }
+        }
+
+        public override IEnumerable<DownloadClientItem> GetItems()
+        {
+            NzbVortexQueue vortexQueue;
+
+            try
+            {
+                vortexQueue = _proxy.GetQueue(30, Settings);
+            }
+            catch (DownloadClientException ex)
+            {
+                _logger.Warn("Couldn't get download queue. {0}", ex.Message);
+                return Enumerable.Empty<DownloadClientItem>();
+            }
+
+            var queueItems = new List<DownloadClientItem>();
+
+            foreach (var vortexQueueItem in vortexQueue.Items)
+            {
+                var queueItem = new DownloadClientItem();
+
+                queueItem.DownloadClient = Definition.Name;
+                queueItem.DownloadId = vortexQueueItem.AddUUID ?? vortexQueueItem.Id.ToString();
+                queueItem.Category = vortexQueueItem.GroupName;
+                queueItem.Title = vortexQueueItem.UiTitle;
+                queueItem.TotalSize = vortexQueueItem.TotalDownloadSize;
+                queueItem.RemainingSize = vortexQueueItem.TotalDownloadSize - vortexQueueItem.DownloadedSize;
+                queueItem.RemainingTime = null;
+                
+                if (vortexQueueItem.IsPaused)
+                {
+                    queueItem.Status = DownloadItemStatus.Paused;
+                }
+                else switch (vortexQueueItem.State)
+                {
+                    case NzbVortexStateType.Waiting:
+                    case NzbVortexStateType.WaitingForSave:
```

Comment:
Thanks @GerTeunis I'll switch that over to downloading, do the other states make sense mapped to Sonarr's status?